backend/app/Providers/AppServiceProvider.php

```diff
@@ -23,6 +25,6 @@ public function register()
       */
      public function boot()
      {
-         //
+         Transaction::observe(TransactionObserver::class);
```

Comment:
prefer EVENT instead of OBSERVER as currently there is only one event,

from the documentation, `If you are listening for many events on a given model, you may use observers to group all of your listeners into a single class`

no need for refactor, lets accept this